external/boost/CMakeLists.txt.boost.in

```diff
@@ -58,7 +59,7 @@ ExternalProject_Add(boost-download
     GIT_SHALLOW         1
     SOURCE_DIR          "@CMAKE_CURRENT_BINARY_DIR@/external_boost"
     UPDATE_COMMAND      ""
-    CONFIGURE_COMMAND   cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @BOOTSTRAP_CMD@ && cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @B2_CMD@ headers
+    CONFIGURE_COMMAND   cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @BOOTSTRAP_CMD@ && cd "@CMAKE_CURRENT_BINARY_DIR@/external_boost" && @B2_CMD@
```

Comment:
Same here. If we do it like this then we will need to compile boost during cmake init. Please do it only if tests are enabled.